src/nvim/CMakeLists.txt

```diff
@@ -56,6 +56,8 @@ set(VIM_MODULE_FILE ${GENERATED_DIR}/lua/vim_module.generated.h)
 set(LUA_VIM_MODULE_SOURCE ${PROJECT_SOURCE_DIR}/src/nvim/lua/vim.lua)
 set(LUA_SHARED_MODULE_SOURCE ${PROJECT_SOURCE_DIR}/runtime/lua/vim/shared.lua)
 set(LUA_INSPECT_MODULE_SOURCE ${PROJECT_SOURCE_DIR}/runtime/lua/vim/inspect.lua)
+set(LUA_F_MODULE_SOURCE ${PROJECT_SOURCE_DIR}/runtime/lua/vim/F.lua)
+set(LUA_META_MODULE_SOURCE ${PROJECT_SOURCE_DIR}/runtime/lua/vim/_meta.lua)
```

Comment:
If these are being shared with tests perhaps we should move them to a `sharded/` subdirectory. That helps signal what is and isn't intended to be "self contained" so that it can be loaded via require() without any transitive dependencies. It also signals to anyone adding things in a `shared/*.lua` module what the expectations are.

Of course, ideally we wouldn't need a "shared" concept if/when `nvim` because the test-runner instead of `lua`.

Reply:
These are not shared with tests. it is just code that is is expected to be part of the builtin `vim` package even before/without `VIMRUNTIME` is properly initialized .

Reply:
Why not put them in `src/nvim/lua` then?

Reply:
We could change things around if we want to but I'm not sure why it would be helpful to spread things around. if someone was looking for `vim.F` they would look into `runtime/lua/vim/F.lua`, mumble a curse under their breath, and then look at `src/nvim/lua/vim/F.lua`. Also:

>  -- Guideline: "If in doubt, put it in the runtime".

Reply:
> but I'm not sure why it would be helpful to spread things around.

How is it spreading things? Isn't it a bit confusing that compiled things are also present in the runtime files, yet inert: changes to the file won't be reflected when calling `vim.F` ? (The exception being `shared*`, which at least has a common, documented indicator "shared".)

> > -- Guideline: "If in doubt, put it in the runtime".

Yes, but there is no doubt based on the decision tree: item (3) says compiled things go in src/nvim/lua.

Reply:
> changes to the file won't be reflected when calling vim.F ? (The exception being shared*, which at least has a common, documented indicator "shared".)

Except that same is true with _all_ runtime files,  when you use any prefixed install, which is the only officially supported way of running neovim. (running directly from the source dir requires hacks, I _have_ been arguing of making this work by a standardized entry point but I was counter-signaled that time, happy to let things be revised of course).